nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+import org.apache.nifi.processors.standard.db.impl.mysql.event.BinlogEventInfo;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.CommitTransactionEventInfo;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.DeleteRowsEventInfo;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.SchemaChangeEventInfo;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.UpdateRowsEventInfo;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.InsertRowsEventInfo;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.io.BeginTransactionEventWriter;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.io.CommitTransactionEventWriter;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.io.DeleteRowsWriter;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.io.InsertRowsWriter;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.io.SchemaChangeEventWriter;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.io.UpdateRowsWriter;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.util.StringUtils;
+import org.apache.nifi.util.file.classloader.ClassLoaderUtils;
+
+import java.io.IOException;
+import java.net.ConnectException;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.DriverPropertyInfo;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+/**
+ * A processor to retrieve Change Data Capture (CDC) events and send them as flow files.
+ */
+@TriggerSerially
+@InputRequirement(InputRequirement.Requirement.INPUT_FORBIDDEN)
+@Tags({"sql", "jdbc", "cdc", "mysql"})
+@CapabilityDescription("Retrieves Change Data Capture (CDC) events from a MySQL database. CDC Events include INSERT, UPDATE, DELETE operations. Events "
+        + "are output as individual flow files ordered by the time at which the operation occurred.")
+@Stateful(scopes = Scope.LOCAL, description = "Information such as a 'pointer' to the current CDC event in the database is stored by this processor, such "
+        + "that it can continue from the same location if restarted.")
+@DynamicProperties({
```

Comment:
Oops, those are left over from when there was going to be a single processor for all CDC vendors. Turned out to be too painful for users, so instead we'll have a CDC processor for each vendor. Will remove these.